org.eclipse.jdt.ui/ui/org/eclipse/jdt/internal/ui/wizards/buildpaths/BuildPathBasePage.java

```diff
@@ -293,10 +293,10 @@ protected boolean getRootExpansionState(TreeListDialogField<CPListElement> list,
 
 	protected void setRootExpansionState(TreeListDialogField<CPListElement> list, boolean state, boolean isClassPathRoot) {
 		for (CPListElement cpListElement : list.getElements()) {
-			if (cpListElement.isClassPathRootNode() && isClassPathRoot) {
+			if (cpListElement.isClassPathRootNode() && isClassPathRoot && list.getTreeViewer() != null) {
```

Comment:
Can we please move that before the for loop and just `if(list.getTreeViewer() == null) {return;}`

Reply:
Fixed